### PR TITLE
MWAA fixes. Our bugs and AWS' as well :)

### PR DESCRIPTION
**Pull request by ashvjit-prophecy**
https://app.asana.com/0/1201492708519695/1206234694955087/f
https://app.asana.com/0/1201492708519695/1206238250596232/f

### CODE REVIEW

File: src/pbt/entities/project.py [hunk 1]
```diff
@@ -368,4 +368,4 @@ def load_pipeline_folder(self, pipeline_id: str):
             path = ".prophecy/{}".format(pid)
             return Project(path, pid).load_pipeline_folder(path)
         else:
-            return None
```
Comment (pankajb23) [inline src/pbt/entities/project.py:371]: This can back-fire in some case where the check is for non-None.

File: src/pbt/client/airflow/mwaa.py [hunk 2]
```diff
@@ -132,6 +132,20 @@ def _extract_bucket_name_from_arn(self) -> str:
...
+        match = re.search(r'\[.*\]', data)
+        if match:
+            json_part = match.group()
+            try:
+                # Parsing the JSON part
+                return json.loads(json_part)
+            except json.JSONDecodeError as e:
+                print(f"Error decoding JSON: {e}")
+        else:
+            print("No JSON found in the data")
```
Comment (pankajb23) [inline src/pbt/client/airflow/mwaa.py:146]: any return statement here or maybe empty list ?

File: src/pbt/entities/project.py (hunk 1 above)
Comment (ashvjit-prophecy) [reply src/pbt/entities/project.py:371]: Changed all usages from None to empty. Although empty was handled

File: src/pbt/client/airflow/mwaa.py (hunk 2 above)
Comment (ashvjit-prophecy) [reply src/pbt/client/airflow/mwaa.py:146]: Changed code
```    
def _extract_and_load_list_json(self, data):
        # Using regular expression to find the JSON part
        match = re.search(r'\[.*\]', data)
        if match:
            json_part = match.group()
            try:
                # Parsing the JSON part
                return json.loads(json_part)
            except json.JSONDecodeError as e:
                print(f"Error decoding JSON: {e}")
                raise DagListParsingtFailedException(f"Error decoding JSON: {e}", e)
        else:
            print("No JSON found in the data")
            raise DagListParsingtFailedException(f"No JSON data found in dag listing")

**Review by pankajb23 — APPROVED** [review]

**Pull request merged**